### PR TITLE
Small tweaks to get doc deployment working

### PR DESCRIPTION
**Pull request by c42f**
Attempt at fixing automated doc deployment. See #298

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
 *.jl.cov
 *.jl.*.cov
 *.jl.mem
+
+docs/build
+docs/site
```
Comment (tkoolen) [inline .gitignore]: Note the .gitignore in the docs directory.

Comment (mschauer) [reply .gitignore]: So this is not necessary, right?

File: docs/make.jl [hunk 2]
```diff
@@ -12,5 +12,6 @@ makedocs(
         )
 
 deploydocs(
-    repo = "github.com/JuliaArrays/StaticArrays.jl",
+    repo = "github.com/JuliaArrays/StaticArrays.jl.git",
+    julia = "0.6"
 )
```
Comment (mschauer) [inline docs/make.jl]: +`deps = nothing,` because otherwise `deps = Deps.pip("pygments", "mkdocs")` is performed

Comment (mschauer) [reply docs/make.jl]: + `make = nothing,` otherwise `run('mkdocs build')` is performed

**Comment by mschauer** [comment]
@tkoolen I could not figure out reason to add `target = "build"`.

File: .gitignore (hunk 1 above)
Comment (c42f) [reply .gitignore]: Thanks I'll remove this.  Strangely I distinctly remember checking for this... I must have been in the working directory.

File: docs/make.jl (hunk 2 above)
Comment (c42f) [reply docs/make.jl]: Done, thanks.

**Comment by tkoolen** [comment]
See #306. Looks good to me after that.

**Comment by c42f** [comment]
Awesome, thanks @tkoolen!

**Comment by c42f** [comment]
@mschauer regarding the `target="build"` thing - perhaps that's an older option required for the `mkdocs`-based Documenter builds? At this stage I'm inclined to just merge this rather than reading the Documenter source in more detail to figure it out ;-)

**Comment by c42f** [comment]
Docs for master are up at:
https://juliaarrays.github.io/StaticArrays.jl/latest/index.html

Some additional cleanup will probably be required, but we've got something to build on now :)

**Comment by tkoolen** [comment]
Note that you won't get 'stable' docs until you tag a new version ('stable' badge in README.md will continue to be a 404).

**Comment by c42f** [comment]
Yep, that's expected, thanks.

**Pull request merged**